### PR TITLE
Removed time.js legacy

### PR DESCRIPTION
**Pull request by snaterlicious**
As Wikibase switched to back-end parsing a while ago, there is no use in further maintaining the ever broken `time.js` component providing front-end parsing and formatting. This pull request integrates all "vital" parts of the `time.js` component directly into `dataValues.TimeValue` that may be interacted with directly now instead of having to retrieve a `time.Time` object from it first.
Note: ValueView needs to be adjusted (change is ready to be submitted).

https://phabricator.wikimedia.org/T89243

### CODE REVIEW

**Comment by tobijat** [comment]
@thiemowmde @adrianheine @mariushoch

**Comment by JeroenDeDauw** [comment]
2 kloc less to maintain. +1 :)

**Comment by adrianheine** [comment]
Nice.

File: src/values/TimeValue.js [hunk 1]
```diff
@@ -129,6 +248,97 @@ SELF.newFromJSON = function( json ) {
...
+// TODO: Inject configurations...
+/**
+ * Calendar configuration.
+ * @property {Object}
+ * @static
+ * @since 0.7
+ */
+SELF.CALENDARS = {
+	GREGORIAN: {
+		text: 'Gregorian',
```
Comment (thiemowmde) [inline src/values/TimeValue.js]: I would have called this "key" and made it lowercase. But I see that this is not new, it was like this before.

File: src/values/TimeValue.js [hunk 2]
```diff
@@ -35,26 +105,107 @@ var SELF = dv.TimeValue = util.inherit( 'DvTimeValue', PARENT, function( value )
...
 	},
 
 	/**
 	 * @inheritdoc
 	 */
 	equals: function( value ) {
-		if ( !( value instanceof SELF ) ) {
-			return false;
+		if( value instanceof SELF ) {
+			return true;
```
Comment (thiemowmde) [inline src/values/TimeValue.js]: Huh? This is enough to decide if two objects are equal? I think this is not correct but not covered by a test, unfortunately.

**Pull request merged**